DD-757 gha workflow med java 21

aksellk opened the pull request.

aksellk committed:
DD-757 GHA workflow for publisering av pakker java 21.

For publisering av pakker til github maven repo.
Samt kjøre mvn package ved pull request.

nikolasrh commented on maven-plugin/pom.xml:
```diff
@@ -147,6 +147,9 @@
                     <goalPrefix>reststop</goalPrefix>
+                    <webAppProvider>
+                        <class>org.eclipse.jetty.ee10.maven.plugin.JettyWebAppProvider</class>
+                    </webAppProvider>
```
Fjerne?

nikolasrh commented on integration-tests/custom-app/api/pom.xml:
```diff
@@ -29,6 +29,7 @@
         <dependency>
             <groupId>org.kantega.reststop</groupId>
             <artifactId>reststop-api</artifactId>
+            <version>${project.version}</version>
```
Tror ikke denne trengs

nikolasrh commented on pom.xml:
```diff
@@ -42,6 +42,7 @@
 
 
     <properties>
+        <revision>3.12-SNAPSHOT</revision>
```
`4.0.0-SNAPSHOT`?

nikolasrh commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,41 @@
+      - name: Get tag from new release
+        if: github.event_name == 'release'
+        run: echo "RELEASE_TAG=${{ github.event.release.tag_name }}" >> $GITHUB_ENV
+      - name: Get latest release tag
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
+        run: echo "RELEASE_TAG=$(git tag --sort=-creatordate | head -n1)-SNAPSHOT" >> $GITHUB_ENV
```
Bruker versjon fra pom.xml i stedet for å finne siste tag i historikken.

nikolasrh commented on .github/workflows/ci.yaml:
```diff
@@ -0,0 +1,41 @@
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```
Kan fjernes siden vi ikke bruker tag i historikken

nikolasrh commented:
Husk squash!

aksellk committed:
DD-757 fjern mvn-release-plugin

Fordi vi overskriver versjon med property med versjonsnummer som argument til maven commands

aksellk committed:
DD-757 fjern release-sign-artifacts

Fordi den ikke er i bruk
